### PR TITLE
SUS-432 | send only required params in authRequestWithMW()

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/SUS-432

`server.js` `authConnection()` requires just these three query parameters:

```js
var roomId = query.roomId;
var name = query.name;
var sessionKey = query.key;
```

### CODE REVIEW

**Review by sqreek — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 59 issues
* ![MAJOR][MAJOR] 49 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 8 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [controllers.js#L31](https://github.com/Wikia/app/blob/a321fc93f811b81f7923fbbcdcad65c7d6deb9bd/extensions/wikia/Chat2/js/controllers/controllers.js#L31): Remove this logging statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2228)
1. ![MAJOR][MAJOR] [controllers.js#L68](https://github.com/Wikia/app/blob/a321fc93f811b81f7923fbbcdcad65c7d6deb9bd/extensions/wikia/Chat2/js/controllers/controllers.js#L68): Remove the unused function parameter "transport". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L93](https://github.com/Wikia/app/blob/a321fc93f811b81f7923fbbcdcad65c7d6deb9bd/extensions/wikia/Chat2/js/controllers/controllers.js#L93): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. ![MAJOR][MAJOR] [controllers.js#L139](https://github.com/Wikia/app/blob/a321fc93f811b81f7923fbbcdcad65c7d6deb9bd/extensions/wikia/Chat2/js/controllers/controllers.js#L139): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L139](https://github.com/Wikia/app/blob/a321fc93f811b81f7923fbbcdcad65c7d6deb9bd/extensions/wikia/Chat2/js/controllers/controllers.js#L139): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L143](https://github.com/Wikia/app/blob/a321fc93f811b81f7923fbbcdcad65c7d6deb9bd/extensions/wikia/Chat2/js/controllers/controllers.js#L143): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L175](https://github.com/Wikia/app/blob/a321fc93f811b81f7923fbbcdcad65c7d6deb9bd/extensions/wikia/Chat2/js/controllers/controllers.js#L175): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L215](https://github.com/Wikia/app/blob/a321fc93f811b81f7923fbbcdcad65c7d6deb9bd/extensions/wikia/Chat2/js/controllers/controllers.js#L215): Remove the unused function parameter "index". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L230](https://github.com/Wikia/app/blob/a321fc93f811b81f7923fbbcdcad65c7d6deb9bd/extensions/wikia/Chat2/js/controllers/controllers.js#L230): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [controllers.js#L255](https://github.com/Wikia/app/blob/a321fc93f811b81f7923fbbcdcad65c7d6deb9bd/extensions/wikia/Chat2/js/controllers/controllers.js#L255): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: extensions/wikia/Chat2/js/controllers/controllers.js [hunk 1]
```diff
@@ -79,8 +79,7 @@ var NodeChatSocketWrapper = $.createClass(Observable, {
 	},
 
 	authRequestWithMW: function (callback) {
-		this.proxy(callback, this)('name=' + encodeURIComponent(wgUserName) + '&key=' + wgChatKey + '&roomId=' + this.roomId
-			+ '&serverId=' + this.wikiId + '&wikiId=' + this.wikiId);
+		this.proxy(callback, this)('name=' + encodeURIComponent(wgUserName) + '&key=' + window.wgChatKey + '&roomId=' + this.roomId);
```
Comment (mszabo-wikia) [inline extensions/wikia/Chat2/js/controllers/controllers.js]: `wikiId` parameter is not passed, so this won't work

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

File: extensions/wikia/Chat2/js/controllers/controllers.js (hunk 1 above)
Comment (ghost) [reply extensions/wikia/Chat2/js/controllers/controllers.js]: @TK-999 , see @macbre 's comment [here](https://wikia-inc.atlassian.net/browse/SUS-432?focusedCommentId=338920&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-338920).

**Review by ghost — APPROVED** [review]

**Review by mszabo-wikia — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 58 issues
* ![MAJOR][MAJOR] 48 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 8 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [controllers.js#L81](https://github.com/Wikia/app/blob/112be1042dff8a7d400bc19ab783140ee75f8201/extensions/wikia/Chat2/js/controllers/controllers.js#L81): Remove the unused function parameter "transport". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L108](https://github.com/Wikia/app/blob/112be1042dff8a7d400bc19ab783140ee75f8201/extensions/wikia/Chat2/js/controllers/controllers.js#L108): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. ![MAJOR][MAJOR] [controllers.js#L154](https://github.com/Wikia/app/blob/112be1042dff8a7d400bc19ab783140ee75f8201/extensions/wikia/Chat2/js/controllers/controllers.js#L154): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L154](https://github.com/Wikia/app/blob/112be1042dff8a7d400bc19ab783140ee75f8201/extensions/wikia/Chat2/js/controllers/controllers.js#L154): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L158](https://github.com/Wikia/app/blob/112be1042dff8a7d400bc19ab783140ee75f8201/extensions/wikia/Chat2/js/controllers/controllers.js#L158): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L190](https://github.com/Wikia/app/blob/112be1042dff8a7d400bc19ab783140ee75f8201/extensions/wikia/Chat2/js/controllers/controllers.js#L190): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L230](https://github.com/Wikia/app/blob/112be1042dff8a7d400bc19ab783140ee75f8201/extensions/wikia/Chat2/js/controllers/controllers.js#L230): Remove the unused function parameter "index". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L245](https://github.com/Wikia/app/blob/112be1042dff8a7d400bc19ab783140ee75f8201/extensions/wikia/Chat2/js/controllers/controllers.js#L245): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [controllers.js#L270](https://github.com/Wikia/app/blob/112be1042dff8a7d400bc19ab783140ee75f8201/extensions/wikia/Chat2/js/controllers/controllers.js#L270): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L287](https://github.com/Wikia/app/blob/112be1042dff8a7d400bc19ab783140ee75f8201/extensions/wikia/Chat2/js/controllers/controllers.js#L287): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 58 issues
* ![MAJOR][MAJOR] 48 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 8 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [controllers.js#L81](https://github.com/Wikia/app/blob/723f2765e9863eee7267f6e848a9d697e5fa6cab/extensions/wikia/Chat2/js/controllers/controllers.js#L81): Remove the unused function parameter "transport". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L108](https://github.com/Wikia/app/blob/723f2765e9863eee7267f6e848a9d697e5fa6cab/extensions/wikia/Chat2/js/controllers/controllers.js#L108): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. ![MAJOR][MAJOR] [controllers.js#L154](https://github.com/Wikia/app/blob/723f2765e9863eee7267f6e848a9d697e5fa6cab/extensions/wikia/Chat2/js/controllers/controllers.js#L154): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L154](https://github.com/Wikia/app/blob/723f2765e9863eee7267f6e848a9d697e5fa6cab/extensions/wikia/Chat2/js/controllers/controllers.js#L154): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L158](https://github.com/Wikia/app/blob/723f2765e9863eee7267f6e848a9d697e5fa6cab/extensions/wikia/Chat2/js/controllers/controllers.js#L158): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L190](https://github.com/Wikia/app/blob/723f2765e9863eee7267f6e848a9d697e5fa6cab/extensions/wikia/Chat2/js/controllers/controllers.js#L190): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L230](https://github.com/Wikia/app/blob/723f2765e9863eee7267f6e848a9d697e5fa6cab/extensions/wikia/Chat2/js/controllers/controllers.js#L230): Remove the unused function parameter "index". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L245](https://github.com/Wikia/app/blob/723f2765e9863eee7267f6e848a9d697e5fa6cab/extensions/wikia/Chat2/js/controllers/controllers.js#L245): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [controllers.js#L270](https://github.com/Wikia/app/blob/723f2765e9863eee7267f6e848a9d697e5fa6cab/extensions/wikia/Chat2/js/controllers/controllers.js#L270): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L287](https://github.com/Wikia/app/blob/723f2765e9863eee7267f6e848a9d697e5fa6cab/extensions/wikia/Chat2/js/controllers/controllers.js#L287): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 58 issues
* ![MAJOR][MAJOR] 48 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 8 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [controllers.js#L81](https://github.com/Wikia/app/blob/2c2f5b4ca120b9561fefd8dc67f1025dc35dd11a/extensions/wikia/Chat2/js/controllers/controllers.js#L81): Remove the unused function parameter "transport". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L108](https://github.com/Wikia/app/blob/2c2f5b4ca120b9561fefd8dc67f1025dc35dd11a/extensions/wikia/Chat2/js/controllers/controllers.js#L108): Add a "default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)
1. ![MAJOR][MAJOR] [controllers.js#L154](https://github.com/Wikia/app/blob/2c2f5b4ca120b9561fefd8dc67f1025dc35dd11a/extensions/wikia/Chat2/js/controllers/controllers.js#L154): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L154](https://github.com/Wikia/app/blob/2c2f5b4ca120b9561fefd8dc67f1025dc35dd11a/extensions/wikia/Chat2/js/controllers/controllers.js#L154): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L158](https://github.com/Wikia/app/blob/2c2f5b4ca120b9561fefd8dc67f1025dc35dd11a/extensions/wikia/Chat2/js/controllers/controllers.js#L158): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L190](https://github.com/Wikia/app/blob/2c2f5b4ca120b9561fefd8dc67f1025dc35dd11a/extensions/wikia/Chat2/js/controllers/controllers.js#L190): Replace "!=" with "!==". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L230](https://github.com/Wikia/app/blob/2c2f5b4ca120b9561fefd8dc67f1025dc35dd11a/extensions/wikia/Chat2/js/controllers/controllers.js#L230): Remove the unused function parameter "index". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [controllers.js#L245](https://github.com/Wikia/app/blob/2c2f5b4ca120b9561fefd8dc67f1025dc35dd11a/extensions/wikia/Chat2/js/controllers/controllers.js#L245): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. ![MAJOR][MAJOR] [controllers.js#L270](https://github.com/Wikia/app/blob/2c2f5b4ca120b9561fefd8dc67f1025dc35dd11a/extensions/wikia/Chat2/js/controllers/controllers.js#L270): Replace "==" with "===". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AEqEqEq)
1. ![MAJOR][MAJOR] [controllers.js#L287](https://github.com/Wikia/app/blob/2c2f5b4ca120b9561fefd8dc67f1025dc35dd11a/extensions/wikia/Chat2/js/controllers/controllers.js#L287): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**